crates/aptos-telemetry-service/src/remote_config.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::common::NodeType;
+use crate::{auth::with_auth, context::Context, types::auth::Claims};
+use warp::filters::BoxedFilter;
+use warp::{reply, Filter, Rejection, Reply};
+
+pub fn telemetry_log_env(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("config" / "env" / "telemetry-log")
+        .and(warp::get())
+        .and(with_auth(
+            context.clone(),
+            vec![
+                NodeType::Validator,
+                NodeType::ValidatorFullNode,
+                NodeType::PublicFullNode,
+            ],
+        ))
+        .and(context.filter())
+        .and_then(handle_telemetry_log_env)
+        .boxed()
+}
+
+async fn handle_telemetry_log_env(
+    claims: Claims,
+    context: Context,
+) -> Result<impl Reply, Rejection> {
+    let env: Option<String> = context
+        .log_env_map()
+        .get(&claims.chain_id)
+        .and_then(|inner| inner.get(&claims.peer_id))
+        .cloned();
+    Ok(reply::json(&env))
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use aptos_config::config::PeerSet;
+    use aptos_types::chain_id::ChainId;
+    use aptos_types::PeerId;
+
+    use crate::jwt_auth::create_jwt_token;
+    use crate::tests::test_context;
+    use crate::types::common::NodeType;
+
+    #[tokio::test]
+    async fn test_handle_telemetry_log_env() {
```

Comment:
Can we add a test when a bad (non-parseable) log_level is sent? Let's say we made a typo in the DB, does this panic all nodes?

Reply:
It doesn't panic, but you stop getting logs. I will add a test to aptos_logger.

Reply:
I see. So if we then undo the change, we will again get logs?

Reply:
Your test checks this -- thanks!